core/io/darwin.rs

```diff
@@ -177,3 +236,19 @@ impl File for DarwinFile {
         }
     }
 }
+
+impl Drop for DarwinFile {
```

Comment:
isnt there a mechanism to put this in the file or in a separate trait? not sure if Rust allows

Reply:
Do you mean implementing the Drop trait on the File trait? I'm not sure rust allows that. 

What do you mean by a separate trait?

Reply:
Is the suggestion directed towards eliminating the duplication of the Drop trait implementation? In which case we could do something like: 

```
pub struct AutoUnlockFile<T: File> {
    inner: T,
}

impl<T: File> AutoUnlockFile<T> {
    pub fn new(file: T) -> Self {
        AutoUnlockFile { inner: file }
    }
}

impl<T: File> Deref for AutoUnlockFile<T> {
    type Target = T;

    fn deref(&self) -> &Self::Target {
        &self.inner
    }
}

impl<T: File> DerefMut for AutoUnlockFile<T> {
    fn deref_mut(&mut self) -> &mut Self::Target {
        &mut self.inner
    }
}

impl<T: File> Drop for AutoUnlockFile<T> {
    fn drop(&mut self) {
        // Ignore errors during drop, as we can't do much about them
        let _ = self.inner.unlock_file();
    }
}
``` 

However not the Windows implementation doesn't implement lock/unlock yet.

Reply:
I might have misinterpreted the original comment! Let me know if that's the case :)

Reply:
may be overcomplicating things